src/discovery.jl

```diff
@@ -154,24 +154,25 @@ next_transition_instant
 
 function next_transition_instant(zdt::ZonedDateTime)
     tz = zdt.timezone
-    tz isa VariableTimeZone || return nothing
+    tz isa AbstractVariableTimeZone || return nothing
 
+    transits = transitions(tz)
```

Comment:
It might be premature optimization to pull this out to be done before.
Idea is to avoid relooking up the `backing_timezone` for `IANATimeZone` multiple times